tests/code-splitting/subtests/dynamic-import-preloading/index.md

```diff
@@ -0,0 +1,63 @@
+---
+title: Dynamic import preloading
+importance: 1
+shortDesc: 'Can lazy-loaded split points be preloaded?'
+---
+
+# Introduction
+
+When creating split points [using dynamic import](/code-splitting/dynamic-import/), it is possible to construct a "waterfall" scenario where JavaScript bundles depend on additional bundles discovered once executed. The ability to preload nested dependency bundles is important in these cases, since doing so can allow the dependency bundles to be downloaded and evaluated in parallel.
+
+# The Test
+
+This test checks to see if it's possible to request code-splitted dependency bundles before the consuming bundle has been loaded, either via [preload](https://web.dev/preload-critical-assets/) or async script loading.
+
+**index.js**
+
+```js
+(async () => {
+  const { shout } =
+    Math.random() < 0.5
+      ? await import('./utilsA.js')
+      : await import('./utilsB.js');
```

Comment:
```suggestion
	// `utilsA` and `utilsB` share a dependency.
	// One import is conditional to make sure the imports and the
	// common dependency stay separate files and will cause a 
	// waterfall loading pattern when loaded naïvely.
    import('./utilsA.js')
    if (Math.random() < 0.5) {
      import('./utilsB.js');
    }
```

@AjayPoshak @developit What do you think about this as a variant/explanation to address the comment above?

Reply:
(if this is okay with both of you, can you  adjust the samples accordingly?)

Reply:
Originally I was going to comment that this seems to make the loading order deterministic, but actually I think that makes the test easier to verify. The correct outcome is that A is always executed prior to B, and in the test this can be verified by execution of B never happening before A.

Otherwise I was going to suggest having the random check swap the order of both imports.

Reply:
I don’t think execution order is well-defined here, since I am not `await`’ing.

Reply:
I guess what I am saying: Both execution orders are valid, which I like because we only want to test if loading happens in parallel. Or am I misunderstanding you?

Reply:
@surma Should I change make changes according to this snippet?

Reply:
Yes I think so, but I am not quite sure if @developit agrees :D

Reply:
Lets merge this PR for now, we can move this discussion to issues, and once we have consensus, we can open a PR to address it.  WDYT?? @surma

Reply:
I’d definitely want an explanatory comment in the code sample before switching this live. Happy to defer the discussion _which_ example is better to a separate issue, tho.

Reply:
I'm happy with the commented version as suggested here. TBH I'm not 100% sure what I was trying to say in my earlier comment. I think it may have been related to order preservation when using hoisted `import()` statements rather than preloading them via `<link>`, since that changes execution order whereas preload does not.

Reply:
Cheers Jason! 

@AjayPoshak Alright! Home stretch! Do you mind taking the suggestion above and adjusting the samples? Then we can merge this 🎉

Reply:
sure, I am on it.

Reply:
I have pushed my changes. Please review.